src/scripts/background/connection/callbacks/ProtocolCallbacks.ts

```diff
@@ -17,10 +20,45 @@ async function addWebpage([url]: [string]): Promise<void> {
   await getMintingRegistrar().maybeTryRegister();
 }
 
+async function isWhitelisted(): Promise<boolean> {
+  return new Promise((resolve, reject) => {
+    chrome.tabs.query(
+      { currentWindow: true, active: true },
+      function (tabArray) {
+        let currentUrl = new URL(tabArray[0].url as string);
+        if (FACT_ACCESS_WHITELIST.includes(currentUrl.host)) {
+          resolve(true);
+        } else {
+          reject("Access denied.");
+        }
+      },
+    );
+  });
+}
+
+async function getTotalFactsCount(): Promise<number> {
+  if (!(await isWhitelisted())) {
+    return 0;
+  }
```

Comment:
Either have `isWhitelisted` return true/false and throw an error in this function instead of returning; or change `isWhitelisted` to `requireWhitelisted` and just call that function plainly.

```
await requireWhitelisted();
...
```

Same for `getFact`.